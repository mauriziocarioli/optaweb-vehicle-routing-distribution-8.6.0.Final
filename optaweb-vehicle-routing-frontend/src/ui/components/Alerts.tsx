/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Alert, AlertActionCloseButton, AlertGroup } from '@patternfly/react-core';
import * as React from 'react';
import { connect } from 'react-redux';
import { messageActions, messageSelectors } from 'store/message';
import { Message } from 'store/message/types';
import { AppState } from 'store/types';

interface StateProps {
  messages: Message[];
}

const mapStateToProps = ({ messages }: AppState): StateProps => ({
  messages: messageSelectors.getNewMessages(messages),
});

interface DispatchProps {
  readMessage: typeof messageActions.readMessage;
}

const mapDispatchToProps: DispatchProps = {
  readMessage: messageActions.readMessage,
};

export type Props = StateProps & DispatchProps;

export const Alerts: React.FC<Props> = ({ messages, readMessage }: Props) => (
  messages.length > 0 ? (
    <AlertGroup isToast>
      {messages
        .map(({ id, text }) => (
          <Alert
            key={id}
            variant="danger"
            title="Error"
            isLiveRegion
            action={(
              <AlertActionCloseButton
                title="Close alert"
                onClose={() => readMessage(id)}
              />
            )}
          >
            {text}
          </Alert>
        ))}
    </AlertGroup>
  ) : null
);

export default connect(mapStateToProps, mapDispatchToProps)(Alerts);
