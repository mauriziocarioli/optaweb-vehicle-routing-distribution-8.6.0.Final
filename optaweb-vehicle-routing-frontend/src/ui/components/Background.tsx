/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { BackgroundImage, BackgroundImageSrc } from '@patternfly/react-core';
import filter from '@patternfly/react-core/dist/styles/assets/images/background-filter.svg';
import pfBackground1200 from '@patternfly/react-core/dist/styles/assets/images/pfbg_1200.jpg';
import pfBackground576 from '@patternfly/react-core/dist/styles/assets/images/pfbg_576.jpg';
import pfBackground1152 from '@patternfly/react-core/dist/styles/assets/images/pfbg_576@2x.jpg';
import pfBackground768 from '@patternfly/react-core/dist/styles/assets/images/pfbg_768.jpg';
import pfBackground1536 from '@patternfly/react-core/dist/styles/assets/images/pfbg_768@2x.jpg';
import React from 'react';

const bgImages = {
  [BackgroundImageSrc.xs]: pfBackground576,
  [BackgroundImageSrc.sm]: pfBackground768,
  [BackgroundImageSrc.xs2x]: pfBackground1152,
  [BackgroundImageSrc.lg]: pfBackground1200,
  [BackgroundImageSrc.sm2x]: pfBackground1536,
  [BackgroundImageSrc.filter]: `${filter}#image_overlay`,
};

const Background: React.FC = () => (
  <BackgroundImage src={bgImages} />
);

export default Background;
