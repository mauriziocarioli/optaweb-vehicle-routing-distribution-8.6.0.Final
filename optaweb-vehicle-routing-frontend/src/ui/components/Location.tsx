/*
 * Copyright 2018 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Button, DataListCell, DataListItem, DataListItemRow, Tooltip,FormGroup,FormSelect,FormSelectOption } from '@patternfly/react-core';
import { TimesIcon } from '@patternfly/react-icons';
import * as React from 'react';
import { LatLangWithId } from 'store/route/types';

export interface LocationProps {
  id: number;
  description: string | null;
  requiredSkill: string;
  lat: number;
  lng: number;
  removeDisabled: boolean;
  removeHandler: (id: number) => void;
  selectHandler: (id: number) => void;
  updateHandler: (value: LatLangWithId,value1: any) => void;
}

const Location: React.FC<LocationProps> = ({
  id,
  description,
  requiredSkill,
  lat,
  lng,
  removeDisabled,
  removeHandler,
  selectHandler,
  updateHandler
}) => {
  const [clicked, setClicked] = React.useState(false);

  var options = [
    { value: 'blood preservation', label: 'Blood Preservation', disabled: false },
    { value: 'blood transfusion', label: 'Blood Transfusion', disabled: false },
    { value: 'depot', label: 'Depot', disabled: false },
    { value: 'vaccination', label: 'Vaccination', disabled: false },
    { value: 'testkit', label: 'Testkit', disabled: false }
    
  ];

  var latlangLoc : LatLangWithId = {
     locationid : id,
     description : description? description : "",
     requiredSkill : requiredSkill,
     lat : lat,
     lng: lng
  }

  function shorten(text: string) {
    const first = text.replace(/,.*/, '').trim();
    const short = first.substring(0, Math.min(20, first.length)).trim();
    if (short.length < first.length) {
      return `${short}...`;
    }
    return short;
  }

  return (
    <DataListItem
      isExpanded={false}
      aria-labelledby={`location-${id}`}
      onMouseEnter={() => selectHandler(id)}
      onMouseLeave={() => selectHandler(NaN)}
    >
      <DataListItemRow>
        <DataListCell isFilled>
          
          { (description && (
            <Tooltip content={description}>
              <span id={`location-${id}`}>{shorten(description)}</span>
            </Tooltip>
          ))
          || <span id={`location-${id}`}>{`Location ${id}`}</span>}
        </DataListCell>
        <DataListCell isFilled>
          <FormGroup label="Skill" fieldId="horizontal-form-title" >
              <FormSelect value={requiredSkill} onChange={ (e : any) => {updateHandler(latlangLoc,e)}}>
                {
                   options.map(function(option,index) {
                        return <FormSelectOption key={index} value={option.value} label={option.label} />
                    })
                }   
              </FormSelect>
          </FormGroup>
        </DataListCell>
        <DataListCell isFilled={false}>
          <Button
            type="button"
            variant="link"
            isDisabled={removeDisabled || clicked}
            onClick={() => {
              setClicked(true);
              removeHandler(id);
            }}
          >
            <TimesIcon />
          </Button>
        </DataListCell>
      </DataListItemRow>
    </DataListItem>
  );
};

export default Location;
