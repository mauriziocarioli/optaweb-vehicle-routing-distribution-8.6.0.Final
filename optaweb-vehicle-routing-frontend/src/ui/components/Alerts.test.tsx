/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Alert } from '@patternfly/react-core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { ReactElement } from 'react';
import { Alerts, Props } from 'ui/components/Alerts';

describe('Alerts', () => {
  it('should call readMessage() when alert is closed', () => {
    const props: Props = {
      messages: [
        { id: '1', text: 'msg 1', status: 'new' },
        { id: '2', text: 'msg 2', status: 'new' },
      ],
      readMessage: jest.fn(),
    };
    const alerts = shallow(<Alerts {...props} />);
    expect(toJson(alerts)).toMatchSnapshot();

    (alerts.find(Alert).at(1).prop('action') as ReactElement).props.onClose();

    expect(props.readMessage).toHaveBeenCalledWith('2');
  });

  it('should not render if there are no messages', () => {
    const props: Props = {
      messages: [],
      readMessage: jest.fn(),
    };
    const alerts = shallow(<Alerts {...props} />);
    expect(toJson(alerts)).toMatchSnapshot();
  });
});
