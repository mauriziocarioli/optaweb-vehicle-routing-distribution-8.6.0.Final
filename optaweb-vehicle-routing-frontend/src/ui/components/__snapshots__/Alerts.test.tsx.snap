// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts should call readMessage() when alert is closed 1`] = `
<AlertGroup
  isToast={true}
>
  <Component
    action={
      <AlertActionCloseButton
        onClose={[Function]}
        title="Close alert"
      />
    }
    isLiveRegion={true}
    key="1"
    title="Error"
    variant="danger"
  >
    msg 1
  </Component>
  <Component
    action={
      <AlertActionCloseButton
        onClose={[Function]}
        title="Close alert"
      />
    }
    isLiveRegion={true}
    key="2"
    title="Error"
    variant="danger"
  >
    msg 2
  </Component>
</AlertGroup>
`;

exports[`Alerts should not render if there are no messages 1`] = `""`;
